modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/main/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtil.java

```diff
@@ -70,13 +66,26 @@ public static List<File> deploy(File lpkgFile, Properties properties)
 					return null;
 				}
 
-				Path lpkgPath = deployerDirPath.resolve(name);
+				innerLPKGFileNames.add(name);
+			}
+		}
+
+		Path deployerDirPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_MARKETPLACE_DIR);
+
+		File deployerDirFile = deployerDirPath.toFile();
+
+		FileUtil.unzip(lpkgFile, deployerDirPath.toFile());
+
+		List<File> lpkgFiles = new ArrayList<>();
+
+		for (String innerLPKGName : innerLPKGFileNames) {
+			File innerLPKGFile = new File(deployerDirFile, innerLPKGName);
```

Comment:
I use a String ArrayList to store all the lpkg inside the lpkgfile, however, which may include the bad lpkg name like "../bad.lpkg", because we do not have the zip slip checker at this moment. After the checker in FileUtil.unzip, it will filter out the zip slip lpkg file, and we won't see this in the marketplace folder. At this moment, we can add the valid lpkg names to the lpkgFiles list